addons/resource/models/resource.py

```diff
@@ -536,193 +476,175 @@ def _schedule_days(self, cr, uid, id, days, day_date=None, compute_leaves=False,
+    hour_to = fields.Float(string='Work to', required=True)
+    calendar_id = fields.Many2one(
+        "resource.calendar", string="Resource's Calendar", required=True)
+
 
 def hours_time_string(hours):
     """ convert a number of hours (float) into a string with format '%H:%M' """
     minutes = int(round(hours * 60))
     return "%02d:%02d" % divmod(minutes, 60)
 
-class resource_resource(osv.osv):
+
+class ResourceResource(models.Model):
     _name = "resource.resource"
     _description = "Resource Detail"
-    _columns = {
-        'name': fields.char("Name", required=True),
-        'code': fields.char('Code', size=16, copy=False),
-        'active' : fields.boolean('Active', track_visibility='onchange',
-            help="If the active field is set to False, it will allow you to hide the resource record without removing it."),
-        'company_id' : fields.many2one('res.company', 'Company'),
-        'resource_type': fields.selection([('user','Human'),('material','Material')], 'Resource Type', required=True),
-        'user_id' : fields.many2one('res.users', 'User', help='Related user name for the resource to manage its access.'),
-        'time_efficiency' : fields.float('Efficiency Factor', size=8, required=True, help="This field depict the efficiency of the resource to complete tasks. e.g  resource put alone on a phase of 5 days with 5 tasks assigned to him, will show a load of 100% for this phase by default, but if we put a efficiency of 200%, then his load will only be 50%."),
-        'calendar_id' : fields.many2one("resource.calendar", "Working Time", help="Define the schedule of resource"),
-    }
-
-    _defaults = {
-        'resource_type' : 'user',
-        'time_efficiency' : 1,
-        'active' : True,
-        'company_id': lambda self, cr, uid, context: self.pool.get('res.company')._company_default_get(cr, uid, 'resource.resource', context=context)
-    }
-
-    def copy(self, cr, uid, id, default=None, context=None):
+
+    name = fields.Char(required=True)
+    code = fields.Char(size=16, copy=False)
+    active = fields.Boolean(default=True, help="If the active field is set to False, it will allow you to hide the resource record without removing it.")
+    company_id = fields.Many2one('res.company', string='Company', default=lambda self: self.env.user.company_id)
+    resource_type = fields.Selection([('user', 'Human'), ('material', 'Material')], string='Resource Type', default='user', required=True)
+    user_id = fields.Many2one('res.users', string='User', help='Related user name for the resource to manage its access.')
+    time_efficiency = fields.Float(string='Efficiency Factor', default=1, size=8, required=True, help="This field depict the efficiency of the resource to complete tasks. e.g  resource put alone on a phase of 5 days with 5 tasks assigned to him, will show a load of 100% for this phase by default, but if we put a efficiency of 200%, then his load will only be 50%.")
```

Comment:
Keeping size attribute will do nothing good here, I suppose. And, not important, but you can write like default=1.0 for the sake of depicting a default value for Float/Monetary fields.